statsd_exporter/Dockerfile

```
@@ -0,0 +1,35 @@
+# syntax=docker/dockerfile:1.8.1@sha256:e87caa74dcb7d46cd820352bfea12591f3dba3ddc4285e19c7dcd13359f7cefd
+
+# Ignoring DL3038 globally because set `assumeyes=True` in dnf.conf in our base image
+# Ignoring DL4006 globally because we inherit the SHELL from our base image
+# hadolint global ignore=DL3038,DL4006
+
+# Not tagging base image because this is an intermediate image
+# hadolint ignore=DL3006
+FROM stackable/image/stackable-base
+ARG PRODUCT
+
+# We don't pin versions
+# hadolint ignore=DL3041
+RUN microdnf update && \
+    microdnf install \
+        # used to unpack the statsd_exporter source
+        tar \
+        gzip \
+        # used to build statsd_exporter
+        golang && \
+    microdnf clean all && \
+    rm -rf /var/cache/yum && \
+    mkdir -p /statsd_exporter
+WORKDIR /statsd_exporter
+
+# hadolint ignore=DL3003
+RUN --mount=type=cache,id=go-statsd-exporter,uid=1000,target=/go_cache <<EOF
+export GOPATH=/go_cache
+curl --fail -L "https://repo.stackable.tech/repository/packages/statsd_exporter/statsd_exporter-${PRODUCT}.src.tar.gz" | tar -xzC .
+(
+    cd "statsd_exporter-${PRODUCT}" || exit
+    go build -o ../statsd_exporter
+)
```

Not sure why the `exit` and the parens were needed...

```suggestion
    cd "statsd_exporter-${PRODUCT}"
    go build -o ../statsd_exporter
```

---

That's the subshell, so the `cd` only affects the commands in the subshell (only `go build` in that case). The commands after the parentheses are executed in the directory we were in before (which is needed for the `rm` to work as intended). As mentioned, I only introduced this because hadolint complained. Before I just did `cd ..` after the `go build`.

---

Ah, got it! So the `ignore=DL3003` comment can be removed?